PR: https://github.com/odoo/odoo/pull/99622

From: 6610e7e18362af453a33d10170dc2d42d5a9171c
From: Kevin Baptiste
Date: 2022-09-07 07:23:47

Structural Changes: 6
Total Changes: 16

[IMP] hr_attendance: add camera choice for barcode scanner

Add new settings to allow the user to select which camera to use for the
barcode scanner and to allow/disallow the manual identification of
employees.

task-2973809

closes odoo/odoo#99622

Related: odoo/enterprise#31082
Signed-off-by: Kevin Baptiste <kba@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_attendance/models/hr_employee.py
+++ b/addons/hr_attendance/models/hr_employee.py
@@ -162,6 +162,7 @@ class HrEmployee(models.Model):
         action_message['barcode'] = employee.barcode
         action_message['next_action'] = next_action
         action_message['hours_today'] = employee.hours_today
+        action_message['kiosk_delay'] = employee.company_id.attendance_kiosk_delay * 1000
 
         if employee.user_id:
             modified_attendance = employee.with_user(employee.user_id).sudo()._attendance_action_change()

--- a/addons/hr_attendance/models/res_company.py
+++ b/addons/hr_attendance/models/res_company.py
@@ -13,6 +13,18 @@ class ResCompany(models.Model):
     overtime_company_threshold = fields.Integer(string="Tolerance Time In Favor Of Company", default=0)
     overtime_employee_threshold = fields.Integer(string="Tolerance Time In Favor Of Employee", default=0)
 
+    attendance_kiosk_mode = fields.Selection([
+        ('barcode', 'Barcode / RFID'),
+        ('barcode_manual', 'Barcode / RFID and Manual Selection'),
+        ('manual', 'Manual Selection'),
+    ], string='Attendance Mode', default='barcode_manual')
+    attendance_barcode_source = fields.Selection([
+        ('scanner', 'Scanner'),
+        ('front', 'Front Camera'),
+        ('back', 'Back Camera'),
+    ], string='Barcode Source', default='front')
+    attendance_kiosk_delay = fields.Integer(default=10)
+
     def write(self, vals):
         search_domain = False  # Overtime to generate
         delete_domain = False  # Overtime to delete

--- a/addons/hr_attendance/models/res_config_settings.py
+++ b/addons/hr_attendance/models/res_config_settings.py
@@ -17,6 +17,9 @@ class ResConfigSettings(models.TransientModel):
         string="Tolerance Time In Favor Of Company", readonly=False)
     overtime_employee_threshold = fields.Integer(
         string="Tolerance Time In Favor Of Employee", readonly=False)
+    attendance_kiosk_mode = fields.Selection(related='company_id.attendance_kiosk_mode', readonly=False)
+    attendance_barcode_source = fields.Selection(related='company_id.attendance_barcode_source', readonly=False)
+    attendance_kiosk_delay = fields.Integer(related='company_id.attendance_kiosk_delay', readonly=False)
 
     @api.model
     def get_values(self):
